libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp

```diff
@@ -49,7 +49,7 @@ void RCInput_ZYNQ::_timer_tick()
     // all F's means no samples available
     while((v = *pulse_input) != 0xffffffff) {
         // Hi bit indicates pin state, low bits denote pulse length
-        if(!(v & 0x80000000))
+        if(v & 0x80000000)
```

Comment:
we may need to fix the coding style in this file. There should be spaces after `if`, `else`, `while` etc. Not for this PR though as it was already wrong.